src/commands/handlers/registration.js

```diff
@@ -213,7 +213,11 @@ const handlers = {
                     ) {
                         handler.connection.write('AUTHENTICATE ' + wanted_mechanism);
                         authenticating = true;
+                    } else {
+                        handleSaslFail(handler, 'unsupported_mechanism');
```

Comment:
I'd rather put the error check first, than having it in the else clause but that may be person preference.
Feel free to ignore this if you disagree.

Reply:
I cannot use return within this code because if disconnect is not true, then it would need to continue processing code under it, the current way around allows for comments indicating which part of the check is for SASL 3.1 and 3.2, switching it around to check first would make it more confusing